lib/util/CompatibilityUtil.js

```diff
@@ -0,0 +1,46 @@
+import { isFunction } from 'min-dash';
+
+// Wraps APIs to check:
+// 1) If a callback is passed -> Warn users about callback deprecation.
+// 2) If Promise class is implemented in current environment.
+export function wrapForCompatibility(api) {
+
+  return function() {
+
+    if (!window.Promise) {
+      throw new Error('Promises is not supported in this environment. Polyfill Promises.');
+    }
+
+    var argLen = arguments.length;
+    if (argLen >= 1 && isFunction(arguments[argLen - 1])) {
+
+      var callback = arguments[argLen - 1];
+
+      console.warn(
+        'Passing callbacks to ' + api.name + ' is deprecated and will be removed in a future major release.' +
+        'Please switch to promises, cf. https://bpmn.io/l/moving-to-promises.html'
+      );
+
+      // We want to call the API without the callback.
+      // Arguments is not an Array, but an array like object. That's why
+      // methods like pop() and slice() do not exist within arguments.
+      delete arguments[argLen - 1];
+
+      api.apply(this, arguments).then(function(result) {
```

Comment:
How about if we just use a slice of arguments? `delete` preserves the length of the array and set the item to `undefined`:
![image](https://user-images.githubusercontent.com/28307541/79975730-b930c580-849b-11ea-93f5-2b6e650dd3b4.png)

C.f. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/delete#Deleting_array_elements

Reply:
So then it would be:
```suggestion
      api.apply(this, Array.prototype.slice.call(arguments, 0, -1)).then(function(result) {
```

Reply:
or even better:
```suggestion
      var argsWithoutCallback = Array.prototype.slice.call(arguments, 0, -1);

      api.apply(this, argsWithoutCallback).then(function(result) {
```